src/deepsparse/loggers/build_logger.py

```diff
@@ -325,6 +338,52 @@ def possibly_modify_target_identifiers(
     return data_logging_config
 
 
+def metric_functions_from_predefined(
+    metric_functions: List[MetricFunctionConfig], identifier_prefix: str
+) -> Generator[Tuple[str, List[MetricFunctionConfig]], None, None]:
+    """
+    Given a list of MetricFunctionConfig objects, parse out
+    the information about the pre-defined data logging configuration.
+
+    Every MetricFunctionConfig.func in the `metric_functions` list
+    maps to a set of built-in functions and identifiers that will be
+    retrieved from the data logging registry to yield pre-defined
+    pairs of (target_identifier, List[MetricFunctionConfig]), that
+    are then used to build the data loggers.
+
+    :param metric_functions: A list containing MetricFunctionConfig
+        objects that specify the predefined data logging configuration.
+    :return: A generator object that generates a sequence of
+        (target_identifier, List[MetricFunctionConfig]) tuples.
+    """
+    # get the registry with the pre-defined data logging configurations
+    from deepsparse.loggers.metric_functions.registry import DATA_LOGGING_REGISTRY
```

Comment:
Postponing the process of registration of all the built-ins until this function is called.

Reply:
why is this necessary? does mutation not properly work with the registry. if so, this is a bug, any time a new function is registered, the global registry should be up to date